pybossa/api/__init__.py

```diff
@@ -417,6 +420,92 @@ def task_progress(project_id=None, short_name=None):
     return Response(json.dumps(task_count_dict), mimetype="application/json")
 
 
+@jsonpify
+@login_required
+@blueprint.route('/preferences/<user_name>', methods=['GET'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def get_user_preferences(user_name=None):
+    """API endpoint for loading account user preferences.
+    Returns a JSON object containing the user account preferences.
+    """
+    if not user_name:
+        return abort(404)
```

Comment:
is this required or following check would be good to handle empty user_name. same in update func

Reply:
This has been refactored in the last version. Both get/set check for an empty user_name.

```
    if not user_name:
        return abort(404)
```